testsuite/tests/typing-modules/functors.ml

```diff
@@ -2054,9 +2104,21 @@ Error: The functor application "Set.Make(Set)(A)" is ill-typed.
          Set A
        do not match these parameters:
          functor (Ord : Set.OrderedType) -> ...
-       1. The following extra argument is provided Set : (module Set)
-       2. Modules do not match:
-            A : sig type a = A.a end
+       1. Modules do not match:
+            Set : (module Set)
+          is not included in
+            Set.OrderedType
```

Comment:
My impression is that this error message is also worse than before -- this would be clearer if `A` had type `OrderedType`, which is how I tend to read this example.

Reply:
This case is completely symmetrical? We are going from a `Delete-Change` to `Change-Delete`? If `A` was matching, the path that will be selected would be `Delete-Match` .

Reply:
The module `A` has type `sig type a end` so it's not clear to me how it can be considered more similar to `OrderedType` than `Set` without requiring human-level sensitivity (although it generated a smaller error message when picked, but I don't know why the missing type `t` and `compare` value were not listed before)

Reply:
I agree. I'm not saying that whatever criterion we use should recognize one from the other, just that the new error message is (for humans) slightly worse than before.